src/utils/parse-sns-message.js

```diff
@@ -0,0 +1,62 @@
+const https = require('https');
+const MessageValidator = require('sns-validator');
+const { promisify } = require('util');
+const logger = require('./logging');
+const config = require('../config');
+
+const validator = new MessageValidator();
+
+const parseSNSMessage = async (req) => {
+  let msg;
+  logger.log('SNS message of length', req.body.length, 'arrived, parsing...');
+
+  // First let's try parsing the body. It should be JSON.
+  try {
+    msg = JSON.parse(req.body);
+  } catch (error) {
+    logger.trace('Parsing error: ', error);
+    throw error;
+  }
+
+  // Asynchronously validate and process the message.
+  const validate = promisify(validator.validate).bind(validator);
+
+  try {
+    msg = await validate(msg);
+  } catch (err) {
+    if (config.clusterEnv === 'development') {
+      logger.log('Error was thrown in development, ignoring it as expected.');
+    } else {
+      logger.error('Error validating the SNS response:', err);
+      throw err;
+    }
+  }
+
+  // Handle subscripton and unsubscription automatically.
+  if (config.clusterEnv !== 'development'
+    && (msg.Type === 'SubscriptionConfirmation'
+      || msg.Type === 'UnsubscribeConfirmation')) {
+    https.get(msg.SubscribeURL);
+  }
```

Comment:
Is this part of the code is preparation for something in the future? or is it linked to another repo?

Reply:
Funny, right now I was looking at the exact same part of the MR :-)

Reply:
The part of the code (lines 35 to 39) make sure that when running the API locally we don't try to get messages from the SNS topic on staging or in production, since we don't have sns locally at all.
The part of the code (lines 42 to 60) parse the message, if there was one received and get the socket io object and return them, so that the rest of the API can do whatever it needs to do with them.